### DISCUSSION TITLE
Support for BACpypes3

### DISCUSSION THREAD

**Discussion by mohakbhalla7** [ChristianTremblay/BAC0#420] [Q&A]
Hi @ChristianTremblay 
I've been using BAC0 for BACnet IP on a Raspberry Pi running Buster image which has Python 3.7 installed. I recently upgraded to Bullseye Image which has Python 3.9 and things started to break. I checked on pypi.org that BACpypes only supports Python versions till Python 3.8. BACpypes3, which is under development, would run on Python> 3.8. According to its author, there would be major overhaul changes in the library. So, I wanted to know, would the current version of BAC0 support BACpypes3 or is there any development in the pipeline to support BACpypes3 in the future?

**Comment by sourcebitt** [comment]
I am running BAC0 and BACpypes on RPI 4, Python 3.11.2 as of 5-Jan-2024 with no issues. Which things started to break?

**Reply by bbartling** [reply]
i think python 3.12 which drops the async-core module things break....

**Reply by sourcebitt** [reply]
@bbartling 
His post stated that he used Python 3.9... I have a feeling there is something else was breaking in his case.

Wait if python 3.12 dropped async-core module, BACpypes3 are going to be breaking right?? (I am not using them yet but was planning to migrate at some point)

**Reply by sourcebitt** [reply]
Oh I see asyncore was replaced with asyncio which is what BACpypes3 uses

**Reply by bbartling** [reply]
Yes exactly...

**Reply by bbartling** [reply]
Python 3.9 should be working no problems with BAC0

**Comment by sourcebitt** [comment]
@mohakbhalla7 Let us know what things were breaking exactly since based on what you wrote there should be zero problems with BAC0 and BACpypes in your case. I am sure eventually BAC0 will work with BACpypes3 since one of the contributors to BACpypes3 is Christian himself.

**Comment by ChristianTremblay** [comment]
You can start to play with it using the async branch of BAC0
